include/seqan/graph_types/graph_iterator_outedge.h

```diff
@@ -92,8 +92,10 @@ class Iter<Graph<Directed<TCargo, TGraphSpec> >, GraphIterator<InternalOutEdgeIt
     Iter(TGraph_ const& _graph, TVertexDescriptor_ const v) :
         data_host(&_graph),
         data_source(v),
-        data_edge(getValue(_graph.data_vertex,v))
+        data_edge()
     {
+        if (!empty(_graph.data_vertex))
+            data_edge = getValue(_graph.data_vertex, v);
```

Comment:
what happens if v does not exist? I would at least add an assertion for this case. Maybe thsis happens already inside getValue but I am not sure.

Reply:
I don't know what exaclt you mean. But `v` is an index (integer) so it will exist when called. The question if `_graph.data_vertex` has at least v+1 elements should not be asserted IMO. The developer has to take care of correctly using this.

Reply:
I agree. The new commit tests for an empty graph already in EdgeIterator and the constructors of OutEdgeIterator are equipped with assertions now.